### ISSUE TITLE
Portico plugin from plugin gallery does not seem to be compatible with OJS 3.4

### ISSUE THREAD

**Issue by rshost** [pkp/pkp-lib#10075]
**Describe the bug**
Different issues exists around the Portico plugin for OJS 3.4 installed from the plugin gallery (referencing to portico-v1_2_0-0.tar.gz).

1) Action links under "Website -> Plugin -> Portico", i.e. "Import/Export Data" and "Upgrade" all redirect to the plugin page. No plugin page or upgrade menu is opened.
2) Trying to deposit an issue (via SFTP to ftp.portico.org with username/PW) results in a blank page and nothing in the error log.

**To Reproduce**
1. Install Portico from the plugin gallery
2. Try links on the plugin page
3. Try to deposit an issue (export works)

**What application are you using?**
 OJS 3.4.0-5

**Additional information**
Portico was used with OJS 3.3 without any issues. All plugin setting were transferred during the upgrade process.

**Comment by rshost** [comment]
@asmecher , @jmacgreg ,

It is some time now with no feedback at all.

Could anyone comment whether this is reproducable? Did anyone use this plugin successfully with OJS 3.4?

**Comment by asmecher** [comment]
@rshost, I tested the plugin and did get a PHP error on deposit (along with a blank screen like you describe):
```
[27-Aug-2024 23:43:10 UTC] PHP Fatal error:  Uncaught TypeError: League\Flysystem\PhpseclibV3\SftpConnectionProvider::__construct(): Argument #6 ($port) must be of type int, string given, called in .../plugins/importexport/portico/PorticoExportPlugin.php on line 168 and defined in .../lib/pkp/lib/vendor/league/flysystem-sftp-v3/SftpConnectionProvider.php:32
```
This is due to more recent versions of PHP getting stricter about type checking, I think, or perhaps the flysystem dependency adding typehints where there previously weren't any. I was able to patch it as follows:
```diff
diff --git a/PorticoExportPlugin.php b/PorticoExportPlugin.php
index d19c5a2..89fdb44 100755
--- a/PorticoExportPlugin.php
+++ b/PorticoExportPlugin.php
@@ -155,7 +155,7 @@ class PorticoExportPlugin extends ImportExportPlugin
                 case 'ftp':
                     $adapter = new \League\Flysystem\Ftp\FtpAdapter(\League\Flysystem\Ftp\FtpConnectionOptions::fromArray([
                         'host' => $credentials['hostname'],
-                        'port' => ($credentials['port'] ?? null) ?: 21,
+                        'port' => ((int) $credentials['port'] ?? null) ?: 21,
                         'username' => $credentials['username'],
                         'password' => $credentials['password'],
                         'root' => $credentials['path'],
@@ -171,7 +171,7 @@ class PorticoExportPlugin extends ImportExportPlugin
                             password: !empty($credentials['private_key']) ? null : $credentials['password'],
                             privateKey: $credentials['private_key'] ?? null ?: null, // Convert possible empty string to null
                             passphrase: $credentials['passphrase'] ?? null ?: null, // Convert possible empty string to null
-                            port: $credentials['port'] ?? 22,
+                            port: ((int) $credentials['port'] ?? null) ?: 22,
                         ),
                         $credentials['path'] ?? '/',
                         \League\Flysystem\UnixVisibility\PortableVisibilityConverter::fromArray([
```

Can you confirm...

1. Whether you're sure there are no errors logged?
2. Whether these changes fix the problem for you?

These only address the second point you raise, about the deposit process not working as expected.

**Comment by rshost** [comment]
Hi @asmecher ,

I can confirm your patch solves the issue with the deposit. Many thanks!

Regarding the error log I'm currently not able to verify this again since logging was disabled when I triggered the blank page again before I applied your patch.

**Comment by asmecher** [comment]
Great, I've released a new version to the plugin gallery containing the fix (1.2.0-1).
